Serhii task 3

zaitsev-serhei opened the pull request:
the task was resolved
updated Strategy classes 
updated formatting of the code

zaitsev-serhei committed:
task was completed. Needs to be tested and reviewed

zaitsev-serhei committed:
updated code for executing transactions.
fixed issue with OutOfBounds in the list.
updated Purchase class to fix negative return value.

zaitsev-serhei committed:
updated code formatting with respect to Maven requirements

Rommelua commented on src/main/java/res/fruitTransactions.txt:
```diff
@@ -0,0 +1,9 @@
+type,fruit,quantity
```
Do not put txt files to **java** folder

Rommelua commented on src/main/java/res/fruitTransactions_1.txt:
```diff
@@ -0,0 +1,3 @@
+fruit,quantity
```
Do not put txt files to **java** folder

Rommelua commented on src/main/java/model/MainTest.java:
```diff
@@ -0,0 +1,8 @@
+package model;
+
+public class MainTest {
```
![image](https://user-images.githubusercontent.com/20326858/208869333-f0baf544-a9d6-466d-8b02-376332572026.png)

Rommelua reviewed (changes requested):
![image](https://user-images.githubusercontent.com/20326858/208869651-7d56d3b5-4646-4df1-8621-e695a717d302.png)

**controller** - is a common name for the package with servlet classes

Rommelua commented on src/main/java/controller/InputData.java:
```diff
@@ -0,0 +1,10 @@
+/*
+ * Implementation of this Interface allows input data from a source
+ */
+public interface InputData {
```
Better have FileService interface with methods for reading and writing

zaitsev-serhei committed:
updated functionality for Strategy performing.
updated input/output with java NIO.
updated code formatting with respect to Maven requirements

Rommelua commented on src/main/java/db/Storage.java:
```diff
@@ -0,0 +1,48 @@
+import model.Fruit;
+
+public class Storage {
+    private HashMap<Fruit, Integer> fruitsInStorage;
```
```suggestion
    private Map<Fruit, Integer> fruitsInStorage;
```

Rommelua commented on src/main/java/db/Storage.java:
```diff
@@ -0,0 +1,48 @@
+    }
+
+    public int getFruitAmount(Fruit fruit) {
+        for (Fruit searchFruit:fruitsInStorage.keySet()) {
+            if (fruit.getName().equals(searchFruit.getName())) {
+                return fruitsInStorage.get(getFruitWithName(fruit.getName()));
+            }
+        }
+        return 0;
```
нужно разобраться, как работает мапа

Rommelua commented on src/main/java/db/Storage.java:
```diff
@@ -0,0 +1,48 @@
+        }
+    }
+
+    public Fruit getFruitWithName(String name) {
```
тут не фигурирует количество, при чем тут Storage?

Rommelua commented on src/main/java/db/Storage.java:
```diff
@@ -0,0 +1,48 @@
+        this.fruitsInStorage.put(fruit,amount);
+    }
+
+    public HashMap<Fruit, Integer> getFruitsInStorage() {
+        return this.fruitsInStorage;
+    }
```
```suggestion

```

Rommelua commented on src/main/java/db/Storage.java:
```diff
@@ -0,0 +1,48 @@
+        return null;
+    }
+
+    public void addFruit(Fruit fruit, int amount) {
+        this.fruitsInStorage.put(fruit,amount);
+    }
```
в чем разница между этим и setFruitAmount?

Rommelua commented on src/main/java/model/FruitTransaction.java:
```diff
@@ -0,0 +1,65 @@
+                    return operation;
+                }
+            }
+            return null;
```
лучше бросить exception

Rommelua commented on src/main/java/model/Main.java:
```diff
@@ -0,0 +1,38 @@
+package model;
```
model??

Rommelua commented on src/main/java/model/Main.java:
```diff
@@ -0,0 +1,38 @@
+        List<FruitTransaction> transactionList = parser.parseTransaction(inputTransactionString);
+        //fill the storage with Fruits
+        StorageService storageService = new StorageServiceImpl();
+        Store store = new Store();
```
Не понял, зачем нам этот объект? Склада не достаточно?

Rommelua commented on src/main/java/model/Main.java:
```diff
@@ -0,0 +1,38 @@
+        //fill the storage with Fruits
+        StorageService storageService = new StorageServiceImpl();
+        Store store = new Store();
+        store.setStorage(storageService.createStorage(transactionList));
+        //execute transactions
+        TransactionExecutor transactionExecutor = new TransactionExecutorImpl(store.getStorage());
+        store.setStorage(transactionExecutor.execute(transactionList));
```
Для создания склада нам нужен лист транзакций? Это как строители дома откажутся класть кирпичи, если будущие жильцы не присутствуют)
А потом мы еще этот лист отдельно выполняем... Ту что-то не так)

Rommelua commented on src/main/java/model/Main.java:
```diff
@@ -0,0 +1,38 @@
+        //execute transactions
+        TransactionExecutor transactionExecutor = new TransactionExecutorImpl(store.getStorage());
+        store.setStorage(transactionExecutor.execute(transactionList));
+        ReportCreator reporter = new ReportCreatorImpl();
+        String report = reporter.createReport(store.getStorage());
```
```suggestion
        ReportCreator reporter = new ReportCreatorImpl(storage);
        String report = reporter.createReport();
```
